src/Hashtable.cpp

```diff
@@ -0,0 +1,155 @@
+#include "Hashtable.h"
+#include <iostream>
+#include "../ShareContainer.h"
+#include "../Share.h"
+#include "../HashUtilities.h"
+#include <string>
+
+Hashtable::Hashtable(int numberOfShares)
+{
+    this->tableSize = 2 * numberOfShares;
+    while(!isPrime(tableSize))
+        tableSize++;                // table size is the a prime bigger than 2 * maximum number of shares
+                                    // ensures a load factor of less than 50%
+
+    ContainerTable = new ShareContainer*[tableSize];  // creates array of pointers
+    for(int i = 0; i < tableSize; i++)
```

Comment:
Das ist clever, ich bin nicht sicher ob das in c++ noch ein Ding ist aber dem sollten wir auf jedenfall noch nachgehen. Vielleicht gibts es auch eine einfache Möglichkeit diese Problematik zu generalisieren.